HW0/P4/P4.py

```diff
@@ -66,12 +100,60 @@
     # B = ?
     # C = ?
 
+    B = np.matrix([[bx, 0, 0, 0, 0, 0],
+                  [0, by, 0, 0, 0, 0],
+                    [0, 0, bz, 0, 0, 0],
+                    [0, 0, 0, bvx, 0, 0],
+                    [0, 0, 0, 0, bvy, 0],
+                    [0, 0, 0, 0, 0, bvz]])
+                    
+    C = np.matrix([[rx, 0, 0, 0, 0, 0], [0, ry, 0, 0, 0, 0], [0, 0, rz, 0, 0, 0]])
     # Initial conditions for s0 and Sigma0
+    s0 = np.array([[0], [0], [2], [15], [3.5], [4.0]])
+    sigma0 = 0.01 * np.identity(6) 
+    mt = m.transpose()
+
     # Compute the rest of sk using Eqs (2), (3), (4), and (5)
+    def predictS(s):
+        return np.dot(A, s) + a
+
+    def predictSig(sigma):
+        pt1 = np.dot(np.dot(A,sigma),A.transpose())
+        pt2 = np.dot(B,B.transpose())
+        return np.linalg.inv(pt1+pt2)
+        
+    def updateSig(sigma):
+        return  np.linalg.inv(sigma+np.dot(C.transpose(),C))
+        
+    def updateS(sigma,sigma0,s,mt):
+        pt1 = sigma*(np.dot(sigma0,s.transpose()))
+        pt2 = np.dot(mt,C)
+        return pt1+pt2
```

Comment:
When running this code we get the following error:
ValueError: operands could not be broadcast together with shapes (6,121) (121,6)
Could it be that one of them needs to be transposed?